velox/benchmarks/SimpleArithmetic.cpp

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <folly/Benchmark.h>
+#include <folly/init/Init.h>
+#include "velox/expression/tests/VectorFuzzer.h"
+#include "velox/functions/Registerer.h"
+#include "velox/functions/lib/benchmarks/FunctionBenchmarkBase.h"
+#include "velox/functions/prestosql/Arithmetic.h"
+#include "velox/parse/ExpressionsParser.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec;
+using namespace facebook::velox::test;
+
+namespace {
+
+class SimpleArithmeticBenchmark
+    : public functions::test::FunctionBenchmarkBase {
+ public:
+  SimpleArithmeticBenchmark() : FunctionBenchmarkBase() {
+    registerFunction<functions::MultiplyFunction, double, double, double>(
+        {"multiply"});
+  }
+
+  void setInput(const TypePtr& inputType, const RowVectorPtr& rowVector) {
+    inputType_ = inputType;
+    rowVector_ = rowVector;
+  }
+
+  // Runs `expression` `times` times.
+  void run(const std::string& expression, size_t times) {
+    folly::BenchmarkSuspender suspender;
+    auto exprSet = compileExpression(expression, inputType_);
+    suspender.dismiss();
+
+    size_t count = 0;
+    for (auto i = 0; i < times; i++) {
+      count += evaluate(exprSet, rowVector_)->size();
+    }
+    folly::doNotOptimizeAway(count);
+  }
+
+ private:
+  TypePtr inputType_;
+  RowVectorPtr rowVector_;
+};
+
+SimpleArithmeticBenchmark benchmark;
+
+BENCHMARK(multiply, n) {
```

Comment:
I wonder how folly plays with the n. 
I have it fixed in my benchmarks .This might be better if folly is really smart.
(i.e. increase n until its a no factor).

Reply:
I was reading about this, the n is essentially the problem size. Folly will run it with increasing values of n, until it figures that the results are stable enough to report. The nice property of specifying n is that you can control the benchmark running time (there's a flag where you can tell folly to run each benchmark for at least a second, for instance, and folly will play with the value of n)